client/my-sites/comments/comment-navigation/index.jsx

```diff
@@ -346,29 +356,43 @@ const mapStateToProps = ( state, { commentsPage, siteId } ) => {
 	};
 };
 
-const mapDispatchToProps = ( dispatch, { siteId } ) => ( {
-	changeStatus: ( postId, commentId, status, analytics = { alsoUnlike: false } ) =>
+const mapDispatchToProps = ( dispatch, { siteId, commentsListQuery } ) => ( {
+	changeStatus: ( postId, commentId, status, options = { alsoUnlike: false } ) =>
 		dispatch(
 			withAnalytics(
 				composeAnalytics(
 					recordTracksEvent( 'calypso_comment_management_change_status', {
-						also_unlike: analytics.alsoUnlike,
-						previous_status: analytics.previousStatus,
+						also_unlike: options.alsoUnlike,
+						previous_status: options.previousStatus,
 						status,
 					} ),
 					bumpStat( 'calypso_comment_management', 'comment_status_changed_to_' + status )
 				),
-				changeCommentStatus( siteId, postId, commentId, status )
+				changeCommentStatus( siteId, postId, commentId, status, {
+					...commentsListQuery,
+					progressId: options.progressId,
+					progressTotal: options.progressTotal,
+				} )
 			)
 		),
-	deletePermanently: ( postId, commentId ) =>
+	deletePermanently: ( postId, commentId, options ) =>
 		dispatch(
 			withAnalytics(
 				composeAnalytics(
 					recordTracksEvent( 'calypso_comment_management_delete' ),
 					bumpStat( 'calypso_comment_management', 'comment_deleted' )
 				),
-				deleteComment( siteId, postId, commentId, { showSuccessNotice: true } )
+				deleteComment(
+					siteId,
+					postId,
+					commentId,
+					{ showSuccessNotice: true },
```

Comment:
I missed this before, but signature wise it's a bit weird to have 2 trailing option objects.

Perhaps:
```
{ 
  showSuccessNotice: true, 
  refreshCommentListQuery: { 
    ...commentsListQuery,
    progressId: options.progressId,
    progressTotal: options.progressTotal,
 }
}
```